### DISCUSSION TITLE
Some clarification on manual installation and installing xformers please.

### DISCUSSION THREAD

**Discussion by fallon33** [comfyanonymous/ComfyUI#274] [Q&A]
Attempting a manual install to experience the new features. In regards to sharing the venv with another install:

"I already have another UI for Stable Diffusion installed do I really have to install all of these dependencies?

You don't. If you have another UI installed and working with it's own python venv you can use that venv to run ComfyUI. You can open up your favorite terminal and activate it:

source path_to_other_sd_gui/venv/bin/activate

or on Windows:

With Powershell: "path_to_other_sd_gui\venv\Scripts\Activate.ps1"

With cmd.exe: "path_to_other_sd_gui\venv\Scripts\activate.bat"

And then you can use that terminal to run Comfyui without installing any dependencies. Note that the venv folder might be called something else depending on the SD UI."_

**I have located activate.bat, but what do I do after that?**

Also for installing xformers:
"Nvidia users should install torch and xformers using this command:

pip install torch==1.13.1 torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu117 xformers"_

**Should I change cu117 to cu119? Anything else to change?**

Thank you.

**Comment by virt-10** [comment]
This is what I did to use an existing venv. Make a bat file inside of comfyui and paste the text below. 

############

@echo off
call cd path_to_other_sd_gui\venv\Scripts
echo %cd%
call activate.bat
echo venv activated
call cd path_to_comfy
echo %cd%
call python main.py --xformers

###########

Change path_to_other_sd_gui\venv\Scripts and  path_to_comfy.

examples
call cd C:\stable-diffusion-webui\venv\Scripts
call cd C:\ComfyUI

If it works you can remove "echo %cd%" its just there to make sure you are in the right directory.

**Reply by Luvdraft** [reply]
But anyway I should copy my existing models to model folder, shouldn't I?

**Reply by bitcrusherrr** [reply]
You can use extra_model_paths.yaml file to point to your existing folder with models check the example file: extra_model_paths.yaml.example

**Comment by ejrejr** [comment]
I'm sharing the Automatic1111 venv now w/ Comfy UI, but I wonder: is that wise?

If not so, then I should be running Comfy UI in its own venv. What are the proper steps to establish that and to keep it up to date?

I have to admit, this is an area where the scripting in A1111 has had a lot more focus & effort.

**Reply by ejrejr** [reply]
Mmm, doi, I guess the flow should be:

Once
- Create your ComfyUI venv (like in the cloned repo)
- Ensure that venv is active
- Do what's needed to install pytorch & xformers according to your platform (e.g., NVidia)
- Install the rest with: pip install -r requirements.txt

Each time there-after:
- Ensure that venv is active
- python main.py
